### ISSUE TITLE
Installed but missing plugin can be re-enabled

### ISSUE THREAD

**Issue by netniV** [Cacti/cacti#1938]
![image](https://user-images.githubusercontent.com/9052188/45590572-fe415d00-b932-11e8-89d6-7cb686c44cb4.png)
The enable plugin (green tick) should not be present, yet despite having a missing directory, you can still enable the plugin:

![image](https://user-images.githubusercontent.com/9052188/45590578-1f09b280-b933-11e8-9d28-2dd9712f322a.png)

**Comment by netniV** [comment]
After update:
![image](https://user-images.githubusercontent.com/9052188/45595678-721d4d00-b9a7-11e8-9058-6750575df144.png)

**Comment by pautiina** [comment]
@netniV Now I have all the plugins look like this, although until today's update everything worked
![image](https://user-images.githubusercontent.com/11805503/45618089-3fde1f00-ba7d-11e8-8496-f0aff5237baf.png)

**Comment by pautiina** [comment]
https://github.com/netniV/cacti/commit/f7d421128ccd0f87b6af8b72a4f6d38597cf7ec7#diff-b01535448f52154b9a6c44885b86584cR254
always: `-5` ?

**Comment by netniV** [comment]
Yes, it's easy to miss but if you review the code fully, you will see that the first part reads the directories available and creates the plugin list, updating them with status.  The second part, goes through the table and sees what directories are NOT available, thus sets a status -5.

**Comment by netniV** [comment]
Yes, because we are looping through the plugin table to find ANY configured plugin that does NOT have a directory.  That would be wrong, as we can't load any information about a plugin from one that simply does not exist! 👍 

The first part is purely looking at what directories exist and updating the status/information if it DOES exist.

Give it a whirl, rename one of your plugin directories and you will see a new entry with the bad name (and a status telling you the name is bad), plus you'll see a plugin error for the old one now, because you've removed it and it no longer exists.

I'm assuming at this point that you "think" it may be a problem, rather than actually seeing one?

**Comment by pautiina** [comment]
The problem I showed above in the screenshot. Now I can not disable and enable the plugin.

**Comment by netniV** [comment]
OK, sorry, missed that.  What is in the /plugins/ folder?

**Comment by pautiina** [comment]
```
drwxr-xr-x   9 www  www   512 17 сент. 13:28 .
drwxr-xr-x  19 www  www  2560 17 сент. 13:23 ..
drwxr-xr-x   7 www  www   512  6 апр.   2017 flowview
-rw-r--r--   1 www  www    40 19 авг.   2017 index.php
drwxr-xr-x   6 www  www   512  6 июля  01:14 intropage
drwxr-xr-x  11 www  www   512 27 марта  2017 ipsubnet
drwxr-xr-x   7 www  www   512 31 июля   2017 monitor
drwxr-xr-x   6 www  www   512 22 авг.  13:05 syslog
drwxr-xr-x   9 www  www  1024 29 авг.  11:34 thold
drwxr-xr-x  16 www  www  2048 17 сент. 13:31 weathermap

```

**Comment by netniV** [comment]
Thanks, I can go one better... I apparently have zero plugins for some reason... let me debug and come back to you 👎

**Comment by netniV** [comment]
It is worth noting that on case sensitive file systems, plugin management will now flag the directory as missing if two copies exist (eg, Monitor and monitor).

**Comment by pautiina** [comment]
Resolved. Thanks.
